src/iter/test.rs

```diff
@@ -1608,3 +1608,14 @@ fn check_partition_map() {
     assert_eq!(a, vec![1, 2, 3]);
     assert_eq!(b, "abcxyz");
 }
+
+#[test]
+fn check_repeat_eq() {
+    let v = vec!(4,4,4,4);
+    let mut fours: Vec<_> = repeat(4)
+    .take(4)
+    .zip(v)
+    .collect();
```

Comment:
So basically despite this being infinite in this formulation, when managing the `IndexedParralellIterator` actually limit the size to usize::MAX and reduce as we consume. So we can split where necessary.